packetbeat/protos/http/event.go

```diff
@@ -95,6 +96,8 @@ func synthesizeFullURL(u *ecs.Url, port int64) string {
 	host := u.Domain
 	if port != 80 {
 		host = net.JoinHostPort(u.Domain, strconv.Itoa(int(u.Port)))
+	} else if strings.IndexByte(u.Domain, ':') != -1 {
+		host = "[" + u.Domain + "]"
```

Comment:
This is necessary because Golang's net/url package expects u.Domain already surrounded with square brackets when it's an IPv6, but surrounds it for you when you use `net.JoinHostPort` above.

Also checking for a `:` character to determine if the host its an IPv6 address is what the url package does internally.